pkg/kwokctl/components/jaeger.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package components
+
+import (
+	"sigs.k8s.io/kwok/pkg/apis/internalversion"
+	"sigs.k8s.io/kwok/pkg/log"
+	"sigs.k8s.io/kwok/pkg/utils/format"
+	"sigs.k8s.io/kwok/pkg/utils/net"
+	"sigs.k8s.io/kwok/pkg/utils/version"
+)
+
+// BuildJaegerComponentConfig is the configuration for building a jaeger component.
+type BuildJaegerComponentConfig struct {
+	Binary       string
+	Image        string
+	Version      version.Version
+	Workdir      string
+	BindAddress  string
+	Port         uint32
+	OtlpGrpcPort uint32
+	Verbosity    log.Level
+	ExtraArgs    []internalversion.ExtraArgs
+	ExtraVolumes []internalversion.Volume
+}
+
+// BuildJaegerComponent builds a jaeger component.
+func BuildJaegerComponent(conf BuildJaegerComponentConfig) (component internalversion.Component, err error) {
+	jaegerArgs := []string{"--collector.otlp.enabled=true"}
+	jaegerArgs = append(jaegerArgs, extraArgsToStrings(conf.ExtraArgs)...)
+
+	var volumes []internalversion.Volume
+	volumes = append(volumes, conf.ExtraVolumes...)
+	var ports []internalversion.Port
+
+	inContainer := conf.Image != ""
+	if inContainer {
+		ports = []internalversion.Port{
+			{
+				HostPort: conf.Port,
+				Port:     16686,
+			},
+		}
+	} else {
+		jaegerArgs = append(jaegerArgs,
+			"--query.http-server.host-port="+conf.BindAddress+":"+format.String(conf.Port),
+			"--collector.otlp.grpc.host-port="+net.LocalAddress+":"+format.String(conf.OtlpGrpcPort),
+		)
+	}
+
+	if conf.Verbosity != log.LevelInfo {
+		jaegerArgs = append(jaegerArgs, "--log-level="+log.ToLogSeverityLevel(conf.Verbosity))
```

Comment:
https://github.com/jaegertracing/jaeger/blob/2c41dae85731ee185fd31fe63f21d76fc2c8055a/pkg/es/config/config.go#L343-L351

There doesn't seem to be a `warn`, we need to define a new conversion for this

Reply:
> https://github.com/jaegertracing/jaeger/blob/2c41dae85731ee185fd31fe63f21d76fc2c8055a/pkg/es/config/config.go#L343-L351
> 
> There doesn't seem to be a `warn`, we need to define a new conversion for this

This is `--es.log-level`.

The '--log-level' is normal.  See https://www.jaegertracing.io/docs/1.47/cli/